certbot-apache/certbot_apache/augeasparser.py

```diff
@@ -0,0 +1,192 @@
+""" Tests for ParserNode interface """
+from certbot_apache import interfaces
+
+from acme.magic_typing import Dict, Tuple  # pylint: disable=unused-import, no-name-in-module
+
+
+class AugeasCommentNode(interfaces.CommentNode):
+    """ Augeas implementation of CommentNode interface """
+    ancestor = None
+    comment = ""
+    dirty = False
+    _metadata = dict()  # type: Dict[str, object]
+
+    def __init__(self, comment, ancestor=None):
+        self.comment = comment
+        self.ancestor = ancestor
+
+    def save(self, msg):  # pragma: no cover
+        pass
+
+    # Apache specific functionality
+
+    def get_metadata(self, key):
+        """ Returns a metadata object
+
+        :param str key: Metadata object name to return
+        :returns: Requested metadata object
+        """
+        try:
+            return self._metadata[key]
+        except KeyError:
+            return None
+
+class AugeasDirectiveNode(interfaces.DirectiveNode):
+    """ Augeas implementation of DirectiveNode interface """
+    ancestor = None
+    parameters = tuple()  # type: Tuple[str, ...]
+    dirty = False
+    enabled = True
+    name = ""
+    _metadata = dict()  # type: Dict[str, object]
+
+    def __init__(self, name, parameters=tuple(), ancestor=None):
```

Comment:
This actually depends on how we're going to solve one of the outstanding issues in the Google doc - handling the added `Include` and `Define` directives on command line, but in general, I think we should make it required.

Reply:
Yeah I think having a required `enabled` parameter makes sense. I suppose it could happen automagically inside the object, but whether or not a directive is enabled depends on its context which we presumably already have outside the object when creating it.